libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpMp4aPayloadReader.java

```diff
@@ -0,0 +1,157 @@
+    firstReceivedTimestamp = C.TIME_UNSET;
+    fragmentedSampleSizeBytes = 0;
+    // The start time offset must be 0 until the first seek.
+    startTimeOffsetUs = 0;
+  }
+
+  @Override
+  public void createTracks(ExtractorOutput extractorOutput, int trackId) {
+    trackOutput = extractorOutput.track(trackId, C.TRACK_TYPE_VIDEO);
+    castNonNull(trackOutput).format(payloadFormat.format);
+  }
+
+  @Override
+  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {
+    checkState(firstReceivedTimestamp == C.TIME_UNSET);
+    firstReceivedTimestamp = timestamp;
+  }
+
+  @Override
+  public void consume(
+      ParsableByteArray data, long timestamp, int sequenceNumber, boolean rtpMarker)
+      throws ParserException {
+    checkStateNotNull(trackOutput);
+
+    int sampleOffset = 0;
+    int numSubFrames = getNumOfSubframesFromMpeg4AudioConfig(payloadFormat.fmtpParameters);
+    for (int subFrame = 0; subFrame <= numSubFrames; subFrame++) {
+      int sampleLength = 0;
+
+      /* Each subframe starts with a variable length encoding */
+      for (; sampleOffset < data.bytesLeft(); sampleOffset++) {
+        sampleLength += data.getData()[sampleOffset] & 0xff;
+        if ((data.getData()[sampleOffset] & 0xff) != 0xff) {
+          break;
+        }
+      }
+      sampleOffset++;
+      data.setPosition(sampleOffset);
+
+      // Write the audio sample
+      trackOutput.sampleData(data, sampleLength);
+      sampleOffset += sampleLength;
+      fragmentedSampleSizeBytes += sampleLength;
+    }
+    if (rtpMarker) {
+      long timeUs =
+          toSampleUs(startTimeOffsetUs, timestamp, firstReceivedTimestamp, payloadFormat.clockRate);
+      trackOutput.sampleMetadata(
+          timeUs, C.BUFFER_FLAG_KEY_FRAME, fragmentedSampleSizeBytes, 0, null);
+      fragmentedSampleSizeBytes = 0;
+    }
+  }
+
+  @Override
+  public void seek(long nextRtpTimestamp, long timeUs) {
+    firstReceivedTimestamp = nextRtpTimestamp;
+    fragmentedSampleSizeBytes = 0;
+    startTimeOffsetUs = timeUs;
+  }
+
+  // Internal methods.
+  private static long toSampleUs(
+      long startTimeOffsetUs, long rtpTimestamp, long firstReceivedRtpTimestamp, int sampleRate) {
+    return startTimeOffsetUs
+        + Util.scaleLargeTimestamp(
+        rtpTimestamp - firstReceivedRtpTimestamp,
+        /* multiplier= */ C.MICROS_PER_SECOND,
+        /* divisor= */ sampleRate);
+  }
+
+  /**
+   * Parses an MPEG-4 Audio Stream Mux configuration, as defined in ISO/IEC14496-3. FMTP attribute
+   * contains config which is a byte array containing the MPEG-4 Audio Stream Mux configuration to
+   * parse.
+   *
+   * @param fmtpAttributes The format parameters, mapped from the SDP FMTP attribute.
+   * @return The number of subframes.
+   * @throws ParserException If the MPEG-4 Audio Stream Mux configuration cannot be parsed due to
+   *     unsupported audioMuxVersion.
+   */
+  public static Integer getNumOfSubframesFromMpeg4AudioConfig(
```

Comment:
It seems the number of subframes is constant (it's from `fmtp`). If so please move this logic into `RtspMediaTrack` where we process `fmtp` for other formats. You can pass this number of subframes in from the constructor instead of calculating it every time.

Reply:
Luyuan, would it be okay if this is called the constructor in this file, so that the function remains private to this class and constructor arguments remain consistent with other readers?

Reply:
That sounds good also!

Reply:
I have called this function in onReceivingFirstPacket() function.

Reply:
Why doing it in `onReceivingFirstPacket` but not in the constructor? `payloadFormat` is final and it seems `numberOfSubframes` can be final too

Reply:
I have moved getNumOfSubframesFromMpeg4AudioConfig() function call inside constructor.